### PR TITLE
feat: Adding info about restricted residential service

### PR DESCRIPTION
**Pull request by mtrunkat**

### CODE REVIEW

**Comment by mtrunkat** [comment]
@jancurn - we need to decide about naming - for me the restricted and full does make the most sense. 

CC @mhamas - here is the doc

File: docs/proxy/residential_proxy.md [hunk 1]
```diff
@@ -19,7 +19,31 @@ Residential proxies support [IP address rotation]({{@link proxy.md#ip-address-ro
...
+Apify provides 2 levels of residential proxy:
+
+| Level | Availability | Supported domains | Other limitations |
+|-------|--------------|-------------------|-------------------|
+| Restricted residential proxy | Every user | >500 domains | Requires man in the middle access to connection |
+| Full residential proxy | Enterprise level accounts | Entire web| None |
+
+### Restricted residential proxy
+
+Restricted residential proxy is available for all the users with some limitations.
```
Comment (jancurn) [inline docs/proxy/residential_proxy.md:33]: Restricted ... with some limitation. That's like double restricted :)

File: docs/proxy/residential_proxy.md [hunk 2]
```diff
@@ -19,7 +19,31 @@ Residential proxies support [IP address rotation]({{@link proxy.md#ip-address-ro
...
+
+First of restricted residential proxy supports only certain domains and paths. The pool of 500 domains is increased every month and covers the most frequent use cases.
```
Comment (jancurn) [inline docs/proxy/residential_proxy.md:35]: Typo? "First of all" or "First, ... " ?

**Comment by jancurn** [comment]
> @jancurn - we need to decide about naming - for me the restricted and full does make the most sense.
> 
> CC @mhamas - here is the doc

Let's imagine you're talking to a custom and explaining to them what they need. If you say "full residential proxy" it might not click to them what tham means. How about "unrestricted" ? that's obvious what it means. 

 @m-murasovs what do you think?

**Comment by mhamas** [comment]
Thx @mtrunkat , I added the route to my PR

File: docs/proxy/residential_proxy.md [hunk 3]
```diff
@@ -19,7 +19,31 @@ Residential proxies support [IP address rotation]({{@link proxy.md#ip-address-ro
...
+| Level | Availability | Supported domains | Other limitations |
+|-------|--------------|-------------------|-------------------|
+| Restricted residential proxy | Every user | >500 domains | Requires man in the middle access to connection |
+| Full residential proxy | Enterprise level accounts | Entire web| None |
+
+### Restricted residential proxy
+
+Restricted residential proxy is available for all the users with some limitations.
+
+First of all, restricted residential proxy supports only certain domains and paths. The pool of 500 domains is increased every month and covers the most frequent use cases.
```
Comment (m-murasovs) [inline docs/proxy/residential_proxy.md]: > The pool of 500 domains is increased every month and covers the most frequent use cases.

Does this mean the user starts out with 500 domains and then gets more?

File: docs/proxy/residential_proxy.md [hunk 4]
```diff
@@ -19,7 +19,31 @@ Residential proxies support [IP address rotation]({{@link proxy.md#ip-address-ro
...
+
+The second limitation is that restricted residential proxy uses the man-in-the-middle system to monitor traffic
+and activities and so requires acceptance of an SSL certificate. This is automatically handled by the [Apify SDK](https://sdk.apify.com/) for both Puppeteer and Playwright.
+
+### Full residential proxy
+
+Full residential proxy neither limits the domains you can access nor requires a man-in-the-middle access to traffic.
+However, it's provided only to enterprise-level accounts on a per-request basis and under an additional contract.
+
+[Contact us](https://apify.com/contact) if you would like to use the full residential proxy or for more information.
```
Comment (m-murasovs) [inline docs/proxy/residential_proxy.md:46]: Maybe a `mailto:support@apify.com` link here? We could even set a custom subject for the email

**Review by m-murasovs — CHANGES_REQUESTED** [review]

**Comment by m-murasovs** [comment]
@jancurn yep, 'unrestricted' would work. I think the customers will still need explanations about what the restricted one is, but there's nothing we can do on that :)

File: docs/proxy/residential_proxy.md (hunk 3 above)
Comment (mtrunkat) [reply docs/proxy/residential_proxy.md]: Nope that the new domains are being enabled in the system globally

Comment (m-murasovs) [reply docs/proxy/residential_proxy.md]: So then you'll still be able to crawl 500 different domains but the overall list will be bigger?

File: docs/proxy/residential_proxy.md [hunk 5]
```diff
@@ -19,7 +19,31 @@ Residential proxies support [IP address rotation]({{@link proxy.md#ip-address-ro
...
+
+### Unrestricted residential proxy
+
+Restricted residential proxy is available for all the users with some conditions.
+
+Firstly, restricted residential proxy supports only certain domains and paths. The pool of 500 domains is increased every month and covers the most frequent use cases.
+Any traffic outside this pool of domains will go through [datacenter proxy]({{@link proxy/datacenter_proxy.md}}).
+
+The second limitation is that restricted residential proxy uses the man-in-the-middle system to monitor traffic
+and activities and so requires acceptance of an SSL certificate. This is automatically handled by the [Apify SDK](https://sdk.apify.com/) for both Puppeteer and Playwright.
```
Comment (jancurn) [inline docs/proxy/residential_proxy.md:39]: Can we add a link to the certificate?

Comment (jancurn) [reply docs/proxy/residential_proxy.md:39]: Plus info how can you tell if the connection is MITM or not - on proxy.apify.com page

**Review by m-murasovs — APPROVED** [review]
Caught some HTML in Markdown - controversial! 😄

**Pull request merged**